### PR TITLE
Spring controller solution

### PR DESCRIPTION
**Pull request by esidarapp**

### CODE REVIEW

File: src/main/java/com/example/spring/Application.java [hunk 1]
```diff
@@ -5,9 +5,9 @@
 
 @SpringBootApplication
 public class Application {
-
-	public static void main(String[] args) {
-		SpringApplication.run(Application.class, args);
-	}
+    public static void main(String[] args) {
+        System.out.println("test");
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/Application.java]:
```suggestion
```

File: src/main/java/com/example/spring/exception/DataProcessingException.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.example.spring.exception;
+
+public class DataProcessingException extends RuntimeException {
+    public DataProcessingException(String s) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/exception/DataProcessingException.java]: s is bad naming, rename to informative parameter

File: src/main/java/com/example/spring/exception/EntityNotFoundException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.example.spring.exception;
+
+public class EntityNotFoundException extends RuntimeException {
+    public EntityNotFoundException(String s) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/exception/EntityNotFoundException.java]: the same

File: src/main/java/com/example/spring/service/BookServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find employee "
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/service/BookServiceImpl.java]: Wrong message

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1 +1,10 @@
 spring.application.name=spring
+server.port=8080
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties]: Actually Tomcat uses 8080 by default, so in this particluar project you don't need to specify the port.

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -1 +1,10 @@
...
+spring.datasource.url=jdbc:mysql://localhost:3306/library
+spring.datasource.username=root
+spring.datasource.password=timofeev
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.jpa.show-sql=true
+spring.jpa.hibernate.ddl-auto=update
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties]:
```suggestion
spring.jpa.hibernate.ddl-auto=create-drop
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve it

File: pom.xml [hunk 7]
```diff
@@ -34,6 +44,31 @@
...
+			<groupId>com.h2database</groupId>
+			<artifactId>h2</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-web</artifactId>
+			<version>6.1.6</version>
+		</dependency>
+		<dependency>
```
Comment (Vasul-Pankiv) [inline pom.xml:57]: redundant

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**